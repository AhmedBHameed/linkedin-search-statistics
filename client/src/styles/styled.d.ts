import 'styled-components';

/**
 * For more information whey we need this file @see https://styled-components.com/docs/api#create-a-declarations-file
 */
declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      blue5: string;
      blue50: string;
      blue90: string;
      red: string;
      highlightText: string;
    };
  }
}
