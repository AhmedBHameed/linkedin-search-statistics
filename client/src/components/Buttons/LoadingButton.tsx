import {Button} from 'antd';
import {ButtonType} from 'antd/es/button';
import {ButtonHTMLType} from 'antd/lib/button/button';
import {SizeType} from 'antd/lib/config-provider/SizeContext';
import React from 'react';
import BoldLabel from '../shared/BoldLabel';

interface LoadingButtonProps {
  loading?: boolean;
  icon?: React.ReactNode;
  onClick?: React.MouseEventHandler<HTMLElement>;
  className?: string;
  size?: SizeType;
  htmlType?: ButtonHTMLType;
  disabled?: boolean;
  type?: ButtonType;
  testId?: string;
}

const LoadingButton: React.FC<LoadingButtonProps> = ({
  children,
  className,
  disabled,
  htmlType,
  icon,
  loading,
  onClick,
  size,
  type,
  testId,
}) => (
  <Button
    className={className}
    data-testid={testId}
    disabled={disabled}
    htmlType={htmlType}
    icon={icon}
    loading={loading}
    onClick={onClick}
    size={size}
    type={type || 'primary'}
  >
    <BoldLabel>{children}</BoldLabel>
  </Button>
);

export default LoadingButton;
