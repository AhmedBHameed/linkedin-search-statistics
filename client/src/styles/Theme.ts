/**
 * The number 50 is the scale of the color according to Figma
 * @see https://www.figma.com/file/WCipxn54SiYJSVpZ3uTUW5/BiomeDx-Styleguide-und-Report-%234?node-id=13%3A140
 */

const highlightText = '#ffc069';

const lightTheme = {
  red: 'red',
  blue5: '#e7f2fe',
  blue50: '#047CB3',
  blue90: '#011e2f',
  highlightText,
};

const darkTheme = {
  red: '#ff7777',
  blue5: '#e7f2fe',
  blue50: '#047CB3',
  blue90: '#011e2f',
  highlightText,
};

export type ThemeProps = typeof darkTheme;
export {lightTheme, darkTheme};
