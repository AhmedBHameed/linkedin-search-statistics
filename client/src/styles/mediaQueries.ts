export const sm = 576;
export const md = 768;
export const lg = 992;
export const xl = 1200;
export const xxl = 1600;

const devices = {
  smMediaQuery: `(min-width: ${sm}px)`,
  mdMediaQuery: `(min-width: ${md}px)`,
  lgMediaQuery: `(min-width: ${lg}px)`,
  xlMediaQuery: `(min-width: ${xl}px)`,
  xxlMediaQuery: `(min-width: ${xxl}px)`,
};

export default devices;
