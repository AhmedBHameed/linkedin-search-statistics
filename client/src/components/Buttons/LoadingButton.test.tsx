import {act, fireEvent, render, waitFor} from '@testing-library/react';
import {ThemeProvider} from 'styled-components';

import {lightTheme} from '../../styles/Theme';
import LoadingButton from './LoadingButton';

function renderComponent(
  props: {loading?: boolean; onClick?: () => void} = {}
) {
  const {loading, onClick} = props;
  return render(
    <ThemeProvider theme={{colors: lightTheme}}>
      <LoadingButton
        loading={loading}
        onClick={onClick}
        testId="loading-button"
      />
    </ThemeProvider>
  );
}

describe('LoadingButton', () => {
  it('matches snapshot', () => {
    const {container} = renderComponent();

    expect(container).toMatchSnapshot();
  });

  describe('Button functionality', () => {
    it('shows loading icon when loading property is true', async () => {
      const {container} = renderComponent({loading: true});

      /**
       * SVGs do not render instantly @see https://www.samdawson.dev/article/svg-graph-testing
       */
      await waitFor(() => {
        expect(container.querySelector('svg')).toHaveAttribute(
          'data-icon',
          'loading'
        );
      });
    });

    it('hides loading icon when loading property is false', async () => {
      const {container} = renderComponent({loading: false});

      await waitFor(() => {
        expect(container.querySelector('svg')).not.toBeInTheDocument();
      });
    });

    it('executes onClick function when clicked', () => {
      const mockButtonOnClick = jest.fn();

      const {getByTestId} = renderComponent({onClick: mockButtonOnClick});

      act(() => {
        fireEvent.click(getByTestId('loading-button'));
      });

      expect(mockButtonOnClick).toHaveBeenCalledTimes(1);
    });
  });
});
