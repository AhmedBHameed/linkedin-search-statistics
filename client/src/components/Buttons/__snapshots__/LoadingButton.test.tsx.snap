// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton matches snapshot 1`] = `
<div>
  <button
    class="ant-btn ant-btn-primary"
    data-testid="loading-button"
    type="button"
  >
    <span
      class="sc-bdnxRM bokTqS"
    />
  </button>
</div>
`;
